src/redux/books/bookSlice.js

```diff
@@ -1,50 +1,110 @@
+  try {
+    const response = await axios.get(apiUrl);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  try {
+    const response = await axios.delete(`${apiUrl}/${id}`);
+    const remainingBooks = response.data;
+    await Promise.all(
+      remainingBooks.map((book, i) => axios.put(`${apiUrl}/${book.item_id}`, { item_id: `item${i + 1}` })),
+    );
+    return response;
+  } catch (error) {
+    return error;
+  }
+});
+
+export const postBook = createAsyncThunk('books/postBook', async (book) => {
+  try {
+    const response = await axios.post(apiUrl, book);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
+
 const bookSlice = createSlice({
-  name: 'book',
+  name: 'books',
   initialState,
   reducers: {
-    addBook: (store, action) => ({
-      ...store,
-      books: [...store.books, action.payload],
-    }),
-    removeBook: (store, action) => {
-      const id = action.payload;
-      const updatedBooks = store.books
-        .filter((book) => book.id !== id)
-        .map((book, index) => ({ ...book, id: `item${index + 1}` }));
-
-      return {
-        ...store,
-        books: updatedBooks,
+    addBook: (state, action) => {
+      const book = {
+        item_id: action.payload.item_id,
+        title: action.payload.title,
+        author: action.payload.author,
+        category: action.payload.category,
       };
+      state.books.push(book);
     },
   },
+  extraReducers: (builder) => {
+    builder.addCase(fetchBooks.pending, (state) => {
+      const newState = state;
+      newState.isLoading = true;
+    });
+    builder.addCase(fetchBooks.fulfilled, (state, action) => {
+      const newState = state;
+      const books = action.payload;
+      const newBooks = Object.keys(books).map((key) => ({
+        item_id: key,
+        title: books[key][0].title,
+        author: books[key][0].author,
+        category: books[key][0].category,
+      }));
+      newState.books = newBooks;
+      newState.isLoading = false;
+    });
+    builder.addCase(fetchBooks.rejected, (state) => {
+      const newState = state;
+      newState.isLoading = false;
+      newState.error = true;
+    });
+    builder.addCase(postBook.pending, (state) => {
+      const newState = state;
+      newState.error = false;
+    });
+    builder.addCase(postBook.fulfilled, (state) => {
+      const newState = state;
+      newState.error = false;
+      window.location.reload();
+    });
+    builder.addCase(postBook.rejected, (state) => {
+      const newState = state;
+      newState.error = true;
+    });
+    builder.addCase(removeBook.pending, (state) => {
+      const newState = state;
+      newState.success = false;
+    });
+    builder.addCase(removeBook.fulfilled, (state) => {
+      const newState = state;
+      newState.success = true;
+      window.location.reload();
+    });
```

Comment:
- [ ] Your app reloads the browser on every book addition or removal. This means we're no longer building a single page app with React. The aim of build with React is not avoid page reload for the removal or fetching of data. Kindly remove the reload methods and refactor your app to add books and remove books without reloading the page.